src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -108,7 +105,26 @@ public function process(ContainerBuilder $container)
                 throw new InvalidArgumentException(sprintf('Service "%s" must implement interface "%s".', $id, $interface));
             }
 
-            $definition->addMethodCall('addSubscriberService', array($id, $class));
+            $r = new \ReflectionClass($class);
+            $extractingDispatcher->addSubscriber($r->newInstanceWithoutConstructor());
```

Comment:
rather than using Reflection, I would overwrite the constructor in ``ExtractingEventDispatcher`` to make it a noop.